backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java

```diff
@@ -0,0 +1,384 @@
+                                                  .bind("id", experimentRun.getId())
+                                                  .execute()),
+                                  executor)
+                              .thenApply(unused2 -> experimentRun, executor);
+                        },
+                        executor),
+            executor);
+    /*.thenCompose(
+    experimentRun -> {
+      // TODO: Fix below logic for checking privileges of linked dataset versions
+      */
+    /*if (experimentRun.getDatasetsCount() > 0 && config.populateConnectionsBasedOnPrivileges) {
+      experimentRun = checkDatasetVersionBasedOnPrivileges(experimentRun, true);
+    }*/
+    /*
+          return InternalFuture.completedInternalFuture(experimentRun);
+        },
+        executor)
+    .thenCompose(
+        experimentRun -> {
+          // TODO: Fix populating logic of setVersioned_inputs,
+          // setHyperparameter_element_mappings here
+          return InternalFuture.completedInternalFuture(experimentRun);
+        },
+        executor)*/
+  }
+
+  /**
+   * Convert CreateExperimentRun request to Experiment object. This method generate the
+   * ExperimentRun Id using UUID and put it in ExperimentRun object.
+   *
+   * @param request : CreateExperimentRun request
+   * @param userInfo : current login UserInfo
+   * @return ExperimentRun : experimentRun
+   */
+  private ExperimentRun getExperimentRunFromRequest(
+      CreateExperimentRun request, UserInfo userInfo) {
+
+    /*
+     * Create ExperimentRun entity from given CreateExperimentRun request. generate UUID and put as
+     * id in ExperimentRun for uniqueness.
+     */
+    if (request.getName().isEmpty()) {
+      request = request.toBuilder().setName(MetadataServiceImpl.createRandomName()).build();
+    }
+
+    ExperimentRun.Builder experimentRunBuilder =
+        ExperimentRun.newBuilder()
+            .setId(UUID.randomUUID().toString())
+            .setProjectId(request.getProjectId())
+            .setExperimentId(request.getExperimentId())
+            .setName(ModelDBUtils.checkEntityNameLength(request.getName()))
+            .setDescription(request.getDescription())
+            .setStartTime(request.getStartTime())
+            .setEndTime(request.getEndTime())
+            .setCodeVersion(request.getCodeVersion())
+            .setParentId(request.getParentId())
+            .addAllTags(ModelDBUtils.checkEntityTagsLength(request.getTagsList()))
+            .addAllAttributes(request.getAttributesList())
+            .addAllHyperparameters(request.getHyperparametersList())
+            .addAllArtifacts(request.getArtifactsList())
+            .addAllDatasets(request.getDatasetsList())
+            .addAllMetrics(request.getMetricsList())
+            .addAllObservations(request.getObservationsList())
+            .addAllFeatures(request.getFeaturesList());
+
+    if (request.getDateCreated() != 0L) {
```

Comment:
Need to check the two fields independently

Reply:
PR: https://github.com/VertaAI/modeldb/pull/2194